projects/ngx-location-picker/src/lib/components/ngx-location-picker.component.ts

```diff
@@ -359,48 +374,36 @@ export class NgxLocationPickerComponent implements OnInit, OnDestroy, ControlVal
   getDeviceLocation() {
     this.isLocating = true;
 
-    if (navigator && navigator.geolocation) {
-      this.geoLocate = navigator.geolocation;
-
-      this.geoLocateId = this.geoLocate.watchPosition((position) => {
-        this.isLocating = false;
-        this.pickedLocation = true;
-
-        if (position && position.coords) {
-          this.setLocationDynamically(position.coords.latitude, position.coords.longitude, true);
-        }
-
-        this.cancelGeolocation();
-      }, (error) => {
+    if (this.geoLocateId != null && this.cachedPosition) {
+      if (this.cachedPosition.coords) {
+        this.setLocationDynamically(
+          this.cachedPosition.coords.latitude,
+          this.cachedPosition.coords.longitude,
+          true
+        );
         this.isLocating = false;
```

Comment:
Deze misschien nog in een aparte private function zetten? #dry

Reply:
done